src/guacd-docker/bin/list-dependencies.sh

```diff
@@ -35,19 +35,12 @@ while [ -n "$1" ]; do
     ldd "$1" | grep -v 'libguac' | awk '/=>/{print $(NF-1)}' \
         | while read LIBRARY; do
 
-        # In some cases, the library that's linked against is a hard link
-        # to the file that's managed by the package, which dpkg doesn't understand.
-        # Searching by */basename ensures the package will be found in these cases.
-        LIBRARY_BASENAME=$(basename "$LIBRARY")
```

Comment:
I'm glad we don't need this anymore.

Reply:
Don't worry - that funkiness had to be replaced with new funkiness: the `sed 's/\(.*\)-[0-9]\..*$/\1/'` at the end.

The `apk` package management tool provides a `-W` option to find the package that owns a file, but the names of the packages printed are in a format that `apk` can't consume for installation. For example:

```console
$ docker run --rm -it alpine
/ # apk add libssl1.1
fetch https://dl-cdn.alpinelinux.org/alpine/v3.16/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.16/community/x86_64/APKINDEX.tar.gz
OK: 6 MiB in 14 packages
/ # apk info -W /lib/libssl.so.1.1 
/lib/libssl.so.1.1 is owned by libssl1.1-1.1.1q-r0
/ # apk add libssl1.1-1.1.1q-r0
ERROR: unable to select packages:
  libssl1.1-1.1.1q-r0 (no such package):
    required by: world[libssl1.1-1.1.1q-r0]
/ # apk add libssl1.1
OK: 6 MiB in 14 packages
/ # 
```

Reply:
Oh ok, cool. We don't want _too_ much improvement all at once.

Reply:
Although, that is a pretty gnarly sed command. A comment explaining what it's doing would be nice.

Reply:
Fair - I'll add that.

Reply:
OK - fixed via rebase.